File: Third_Maximum/third_maximum.py
def third_maximum(nums):
    nums = sorted(set(nums), reverse=True)

    if len(nums) >= 3:
        third_max = nums[2]
        return third_max
    else:
        first_max = nums[0]
        return first_max

print(third_maximum([1, 2]))
print(third_maximum([3, 2, 1]))
print(third_maximum([2, 2, 3, 1]))

File: Third_Maximum/thirdMaximum.java
import java.util.*;

public class thirdMaximum{
    public static void main(String[] args) {
        process(new int[] {3,2,1});
        process(new int[] {1,2});
        process(new int[] {2,2,3,1});
    }

    public static void process(int[] nums) {
        TreeSet<Integer> sortedSet = new TreeSet<>(Collections.reverseOrder());

        for (int num : nums) {
            sortedSet.add(num);
        }

        List<Integer> sortedNums = new ArrayList<>(sortedSet);

        if(sortedNums.size() >= 3) {
            System.out.println(sortedNums.get(2));
        } else {
            System.out.println(sortedNums.get(0));
        }
    }
}
